### ISSUE TITLE
"Debugger creation failed. There are grammar errors" with no visible errors

### ISSUE THREAD

**Issue by matthew-dean** [mike-lischke/vscode-antlr4#212]
So, I'm sorry that this may be a hard to repo issue. But, occasionally, I make a parser / lexer change and press "Run and Debug" in VSCode and I get: `Could not prepare debug session: Error: Debugger creation failed. There are grammar errors.` but there will be no errors shown in the Debug console or highlighted in the `.g4` documents.

I guess what might be nice and what I might be able to help pin down the bug for is to somehow identify how / where errors occurred.

Since I'm just learning Antlr 4, I totally assume I wrote something dumb, and thus far, by trial and error, I've been able to make that error go away without understanding the reason, but it would be nice to have more feedback about whereabouts this error might have been triggered.

**Comment by mike-lischke** [comment]
Can you see if there's a pattern, like you changed something which was an error, fixed it and ran the debugger in a quick manner and got that debugger message? Maybe the fixed grammar wasn't ready in time yet, when you started debugging? Something along this line...

**Comment by matthew-dean** [comment]
It's very inconsistent. Here was my experience today:
1. I opened up a project with Antlr files.
2. I ran the debugger I had set up in my project.
3. No errors. No results. No output in the Output console.
4. I commented out a line that I knew was causing me trouble before. The debugger ran, and gave me a parse tree, albeit with some errors.
5. I un-commented out the line and saved. Now it looks identical to when I started. But now I get a parse tree.
6. To try to figure out one of my parse tree errors, I altered one of my tokens.
7. The Output says "The specified token does not exist."
8. I reverted my change in saved. The output says, "The specified token does not exist."

In other words, with the _same lexer and same grammar_ I got 3 different outputs:
1. Nothing
2. A parse tree
3. "The specified token does not exist."

It seems like the extension is preserving _something_ of the state of the previous compilation, because the input does not map consistently to output.

**Comment by matthew-dean** [comment]
I think the main bug I can identify is that this extension does not re-compile / update errors on a lexer until you've saved in a parser. Meaning, if you make changes to a lexer and save it, and that lexer is imported into a `parser grammar` via `{ tokenVocab = CssLexer}`, your lexer will not display errors (or won't in my case) _in the lexer_ until I tab over to `CssParser.g4` and re-save. The same is true for parse tree generation etc. So I had many cases where I had the same exact lexer code but was getting different errors & output when running a launch.json file to debug. So, same input = different output.

I think that's a source of a lot of errors and confusion. Errors are stale, and can refer to old code, or you can see errors in code that was working before. Or you can get no errors and then open VSCode later and have errors. I don't know if that's just a configuration issue but that's been the main reason why I've been confused. And it's possible some of the bugs I think I'm seeing related to this single bug.

**Comment by matthew-dean** [comment]
So, the case of the specific "Debugger creation failed. There are grammar errors" popup error, I recreated it just now by editing a token that referred to a non-existent other token. The actually lexer file shows no errors, _unless_ I re-open / re-save my parser file. If I revert the change while my parser file is open, it still displays an error. If I close the parser file and re-open, or if I tab to the parser file and save, only then does the error go away in my lexer file.
